### PR TITLE
efficient ad

### PR DESCRIPTION
**Pull request by samuel-wj-chapman**
## Pull Request Description:
Addition of anomaly detection model

## Checklist before requesting a review:
- [x] I set the appropriate labels on the pull request.
- [x] I have added/updated the release note draft (if necessary).
- [x] I have updated the documentation to reflect my changes (if necessary).
- [x] All function and files are well documented. 
- [x] All function and classes have type hints.
- [x] There is a licenses in all file.
- [x] The function and variable names are informative. 
- [x] I have checked for code duplications.
- [x] I have added new unittest (if necessary).

### CODE REVIEW

**Comment by Idan-BenAmi** [comment]
Hi Samuel, several comments:
- Dataset link - maybe require a note about mvtec license?
- Function names - keep our coding conventions (function names with snack case, classes with camel case)
- Training - very nice, let's talk if it this is not out of context.
- notebook Text - use US english speller, for alignment (..my apologies :)
- notebook Text - add links (to Mvtec, to efficientAD paper)
- New files/functions - let's talk where to locate the new files
- Link to the new tutorial from imx500 notebooks README

**Comment by samuel-wj-chapman** [comment]
@Idan-BenAmi should be good for review

File: tutorials/notebooks/imx500_notebooks/pytorch/pytorch_efficient_anomaly_detection.ipynb [hunk 1]
```diff
@@ -0,0 +1,824 @@
...
+        "id": "LF92AFYDJGd0"
+      },
+      "source": [
+        "Classification models are powerful and reliable, but what if you have little or no examples of one of your classes, what if one of your classes contains too much unpredictable variation?\n",
+        "\n",
+        "Here we go through the process of building, training and quantising an anomaly detection model designed to solve exactly these problems.\n",
+        "\n",
+        "Anomaly detection models are useful as they only require your typical images to train and can in theory determin anything that is not typical.\n",
+        "\n",
+        "We use Efficient ad, one of the top performing anomaly detection models on the mvtec benchmark. benchmark leader board can be found [here](https://paperswithcode.com/sota/anomaly-detection-on-mvtec-ad)\n",
```
Comment (Idan-BenAmi) [inline tutorials/notebooks/imx500_notebooks/pytorch/pytorch_efficient_anomaly_detection.ipynb:37]: We use *Efficient ad* --> add link
on the *mvtec benchmark* --> add link (if available)

File: tutorials/notebooks/imx500_notebooks/pytorch/pytorch_efficient_anomaly_detection.ipynb [hunk 2]
```diff
@@ -0,0 +1,824 @@
...
+        "sys.path.insert(0,\"tutorials\")"
+      ]
+    },
+    {
+      "cell_type": "markdown",
+      "metadata": {
+        "id": "tGqXJH15oLvm"
+      },
+      "source": [
+        "Download and extract the mvtec benchmark dataset. This is used for both training and evaluation. Link below is a direct linkk from the mcvtec website."
```
Comment (Idan-BenAmi) [inline tutorials/notebooks/imx500_notebooks/pytorch/pytorch_efficient_anomaly_detection.ipynb:121]: direct linkk -- >direct link
from the *mcvtec website* -- >add link

File: tutorials/mct_model_garden/models_pytorch/Efficient_Anomaly_Det/efficient_ad.py [hunk 3]
```diff
@@ -0,0 +1,227 @@
...
+#   * Modify layers to make them more suitable for quantization.
+#   * Inheritance class from HuggingFace
+#   * Uninfied version of model combining the three subversions
+# ==============================================================================
+"""
+Yolov8n Object Detection Model - PyTorch implementation
+
+This code contains a PyTorch implementation of efficient ad model, following
+https://github.com/nelson1425/EfficientAD. This implementation includes a unified version of the model that combines the three submodels 
+into one to ease the process of quantisation and deployment.
```
Comment (Idan-BenAmi) [inline tutorials/mct_model_garden/models_pytorch/Efficient_Anomaly_Det/efficient_ad.py:13]: quantisation --> quantization

File: tutorials/notebooks/imx500_notebooks/pytorch/pytorch_efficient_anomaly_detection.ipynb [hunk 4]
```diff
@@ -0,0 +1,824 @@
...
+    },
+    {
+      "cell_type": "markdown",
+      "metadata": {
+        "id": "LF92AFYDJGd0"
+      },
+      "source": [
+        "Classification models are powerful and reliable, but what if you have little or no examples of one of your classes, what if one of your classes contains too much unpredictable variation?\n",
+        "\n",
+        "Here we go through the process of building, training and quantising an anomaly detection model designed to solve exactly these problems.\n",
```
Comment (Idan-BenAmi) [inline tutorials/notebooks/imx500_notebooks/pytorch/pytorch_efficient_anomaly_detection.ipynb:33]: quantising -->  quantizing

File: tutorials/notebooks/imx500_notebooks/pytorch/pytorch_efficient_anomaly_detection.ipynb [hunk 5]
```diff
@@ -0,0 +1,824 @@
...
+        "!python ./mvtec_ad_evaluation/evaluate_experiment.py --dataset_base_dir './mvtec_anomaly_detection/' --anomaly_maps_dir './output/anomaly_maps/mvtec_ad/' --output_dir './output/metrics/mvtec_ad/' --evaluated_objects bottle"
+      ]
+    },
+    {
+      "cell_type": "markdown",
+      "metadata": {
+        "id": "_fPTNtJLgPc5"
+      },
+      "source": [
+        "## Quantised model benchmark\n"
```
Comment (Idan-BenAmi) [inline tutorials/notebooks/imx500_notebooks/pytorch/pytorch_efficient_anomaly_detection.ipynb:364]: Quantised --> Quantized

File: tutorials/notebooks/imx500_notebooks/pytorch/pytorch_efficient_anomaly_detection.ipynb [hunk 6]
```diff
@@ -0,0 +1,824 @@
...
+    },
+    {
+      "cell_type": "markdown",
+      "metadata": {
+        "id": "n1zwMJyri8by"
+      },
+      "source": [
+        "## Conclusion\n",
+        "\n",
+        "In this notebook we provide examples on how to quantise and benchmark the latest anomaly detection model as well as providing code to visulize the models output."
```
Comment (Idan-BenAmi) [inline tutorials/notebooks/imx500_notebooks/pytorch/pytorch_efficient_anomaly_detection.ipynb:440]: quantise --> quantize

**Review by Idan-BenAmi — APPROVED** [review]

**Pull request merged**